.github/workflows/tests.yml

```diff
@@ -53,71 +56,224 @@ jobs:
 
       - name: Check upstream
         if: github.event_name == 'pull_request' && matrix.runtime-version == 'latest'
-        run: |
-          COMMIT="$(git log -n 1 --pretty=format:%s HEAD^2)"
-          if [[ "$COMMIT" == *"test-upstream"* ]]
-          then
-            export TEST_UPSTREAM="true"
-            # Put TEST_UPSTREAM into $GITHUB_ENV so it can be used in subsequent workflow steps
-            echo TEST_UPSTREAM=$TEST_UPSTREAM >> $GITHUB_ENV
-            echo "Using development version of dask and distributed"
-          fi
+        run: bash ci/scripts/check_upstream.sh
 
       - name: Install coiled-runtime
+        env:
+          COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
+        run: bash ci/scripts/install_coiled_runtime.sh
+
+      - name: Export environment
+        env:
+          COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
+          PYTHON_VERSION: ${{ matrix.python-version }}
```

Comment:
I don't think these environment variables are needed here (or other places we run `export_conda_env.sh`)

```suggestion
```

Reply:
Removed